client/verta/verta/_tracking/experimentrun.py

```diff
@@ -1332,36 +1335,37 @@ def log_model(self, model, custom_modules=None, model_api=None, artifacts=None,
             raise ValueError(
                 "`artifacts` can only be provided if `model` is a class")
 
-        # build model API
-        if model_api is None:
-            model_api = utils.ModelAPI()
-        elif not isinstance(model_api, utils.ModelAPI):
-            raise ValueError(
-                "`model_api` must be `verta.utils.ModelAPI`, not {}".format(type(model_api)))
-        if 'model_packaging' not in model_api:
-            # add model serialization info to model_api
-            model_api['model_packaging'] = {
-                'python_version': _utils.get_python_version(),
-                'type': model_type,
-                'deserialization': method,
-            }
-        if self._conf.debug:
-            print("[DEBUG] model API is:")
-            pprint.pprint(model_api.to_dict())
-
         # associate artifact dependencies
         if artifacts:
             self.log_attribute(_MODEL_ARTIFACTS_ATTR_KEY, artifacts, overwrite)
 
-        custom_modules_artifact = self._custom_modules_as_artifact(
-            custom_modules)
-        self._log_artifact("custom_modules", custom_modules_artifact,
-                           _CommonCommonService.ArtifactTypeEnum.BLOB, 'zip', overwrite=overwrite)
-
+        # create and upload model API
+        if model_type or model_api:  # only if provided or model is deployable
+            if model_api is None:
+                model_api = utils.ModelAPI()
+            if 'model_packaging' not in model_api:
+                # add model serialization info to model_api
+                model_api['model_packaging'] = {
+                    'python_version': _utils.get_python_version(),
+                    'type': model_type,
+                    'deserialization': method,
+                }
+            if self._conf.debug:
+                print("[DEBUG] model API is:")
+                pprint.pprint(model_api.to_dict())
+
+            self._log_artifact("model_api.json", model_api,
```

Comment:
suggest naming this constant string

Reply:
4a8de7719bc9503f34f2f20f407fe7db5c04dec1
2a62500b1514c331b02411bc503be407ac8dcb98